File: scala/src/test/scala/BeveragesPricingTest.scala
import org.scalatest.FunSuite
import org.scalatest.Matchers._


class BeveragesPricingTest extends FunSuite {

  test("computes coffee price"){
    val coffee = new Coffee
    coffee.price() should be (1.20 +- 0.0001)
  }

  test("computes tea price"){
    val tea = new Tea
    tea.price() should be (1.50 +- 0.0001)
  }

  test("computes hot chocolate price"){
    val hotChocolate = new HotChocolate
    hotChocolate.price() should be (1.45 +- 0.0001)
  }

  test("computes tea with milk price"){
    val teaWithMilk = new TeaWithMilk
    teaWithMilk.price() should be (1.60 +- 0.0001)
  }

  test("computes coffee with milk price"){
    val coffeeWithMilk = new CoffeeWithMilk
    coffeeWithMilk.price() should be (1.30 +- 0.0001)
  }

  test("computes coffee with milk and cream price"){
    val coffeeWithMilkAndCream = new CoffeeWithMilkAndCream
    coffeeWithMilkAndCream.price() should be (1.45 +- 0.0001)
  }

  test("computes hot chocolate with cream price"){
    val hotChocolateWithCream = new HotChocolateWithCream
    hotChocolateWithCream.price() should be (1.60 +- 0.0001)
  }

}

File: solutions/dotnet-core/pilar_rubio/Beverages.Test/BeveragesPricingTest.cs
using Xunit;

namespace Beverages.Test
{
    public class BeveragesPricingTest
    {
        private ISupplement _supplement;

        public BeveragesPricingTest()
        {
            _supplement = new Supplement();
        }

        [Fact]
        public void ComputesCoffeePrice()
        {
            var coffee = new Coffee(_supplement);
            Assert.Equal(1.20, coffee.Price(), 3);
        }

        [Fact]
        public void ComputesTeaPrice()
        {
            var tea = new Tea(_supplement);
            Assert.Equal(1.50, tea.Price(), 3);
        }

        [Fact]
        public void ComputesHotChocolatePrice()
        {
            var hotChocolate = new HotChocolate(_supplement);
            Assert.Equal(1.45, hotChocolate.Price(), 3);
        }

        [Fact]
        public void ComputesTeaWithMilkPrice()
        {
            var teaWithMilk = new Tea(_supplement);
            Assert.Equal(1.60, teaWithMilk.PriceWithMilk(), 3);
        }

        [Fact]
        public void ComputesCoffeeWithMilkPrice()
        {
            var coffeeWithMilk = new Coffee(_supplement);
            Assert.Equal(1.30, coffeeWithMilk.PriceWithMilk(), 3);
        }

        [Fact]
        public void ComputesCoffeeWithMilkAndCreamPrice()
        {
            var coffeeWithMilkAndCream = new Coffee(_supplement);
            Assert.Equal(1.45, coffeeWithMilkAndCream.PriceWithMilkAndCream(), 3);
        }

        [Fact]
        public void ComputesHotChocolateWithCreamPrice()
        {
            var hotChocolateWithCream = new HotChocolate(_supplement);
            Assert.Equal(1.60, hotChocolateWithCream.PriceWithCream(), 3);
        }

        [Fact]
        public void ComputesCoffeeWithCinnamonPrice()
        {
            var coffeeWithCinammon = new Coffee(_supplement);
            Assert.Equal(1.25, coffeeWithCinammon.PriceWithCinnamon(), 3);
        }
    }
}
